utils.js

```diff
@@ -0,0 +1,246 @@
+const aws = require('aws-sdk')
+const { equals, head, includes, isNil, map, merge, not, reduce, toPairs } = require('ramda')
+const { utils } = require('@serverless/core')
+const { schema } = require('./schema')
+
+/**
+ * Get AWS clients
+ * @param {object} credentials
+ * @param {string} region
+ * @returns {object} AWS clients
+ */
+const getClients = (credentials, region = 'us-east-1') => {
+  const cloudformation = new aws.CloudFormation({ credentials, region })
+  const s3 = new aws.S3({ credentials, region })
+  return {
+    cloudformation,
+    s3
+  }
+}
+
+/**
+ * Waits CloudFormation stack to reach certain event
+ * @param {object} cloudformation
+ * @param {string} events events to wait for
+ * @param {object} config
+ * @returns {array} stack outputs
+ */
+const waitFor = async (cloudformation, events, config) =>
+  new Promise(async (resolve, reject) => {
+    const inProgress = true
+    do {
+      try {
+        await utils.sleep(5000)
+        const { Stacks } = await cloudformation
+          .describeStacks({ StackName: config.stackName })
+          .promise()
+        if (includes(head(Stacks).StackStatus, events)) {
+          return resolve(Stacks)
+        }
+      } catch (error) {
+        return reject(error)
+      }
+    } while (inProgress)
+  })
```

Comment:
that is such a cool utility! 😍 ... do you think we can decouple it in a way and include it in the core utils?

Reply:
sure, aws-sdk actually has similar functionality build-in to CloudFormation, but the problem with it is that the polling is done every 30 seconds and cannot be adjusted. So the minimum waiting time is 30 seconds regardless of the changes made. This speeds up the deployment a little bit.

Reply:
that is pretty useful. Just like the `sleep` method